pyproject.toml

```diff
@@ -160,7 +160,9 @@ python = ["3.9", "3.11"]
 dependencies = ["tmt[docs]"]
 
 [tool.hatch.envs.docs.scripts]
-html = "make -C {root}/docs html"
+html = "sphinx-build -b html {root}/docs {root}/docs/_build {args}"
+linkcheck = "sphinx-build -b linkcheck {root}/docs {root}/docs/_build {args}"
+doctest = "sphinx-build -b doctest {root}/docs {root}/docs/_build {args}"
```

Comment:
Hm, I'd say -1 from me, sorry :( `make` is really, really nice for tasks that are behind this, tasks like calling scripts and also **not** calling scripts because it's not necessary. There will be more scripts and more content generated from sources, and I see no advantage in turning the implementation into Python code.

Reply:
I don't see `docs/Makefile` going away - there are tasks for which `make` is better suited than Python code. But, after some thinking, it's probably not necessary for `make -C docs/ html` to exist.

If we state that `docs/Makefile` takes care of generating content from sources - and `make` is just better for recognizing when not to bother doing things, which is of no use in Githib action but much appreciated when refactoring docs locally - then we should remove the building targets - `html`, `man`, `text`, `latexpdf` - from `docs/Makefile`, and move the corresponding commands into `pyproject.toml`.

As mentioned above, `conf.py` could call `make -C docs generate` to create the content, leaving the management and decisions to `make`. Then we'd have one way to build various flavors of docs - `hatch run docs:linkcheck`, for example - that calls `sphinx-build`, `conf.py` let's `make` prepare content (possibly doing nothing on my laptop...) and produce docs. That should work in both RTD task, Github actions for new builders and on developer's laptop.

Reply:
One point against the manual `Makefile` is that it is not as intuitive and easy to navigate, but if we can generalize the Makefile to accept arguments like `BUILDER`/`-b` to accept generic builders and then pass everything else transparently, then we can convert them to makefile format.

I think there can be cases where we want to resolve the builder and add some default arguments as well (reference project: https://github.com/scikit-build/scikit-build-core/blob/de72cb49fcf57fd6d2025a60338f5d58272933d1/noxfile.py#L142)

Reply:
Actually, as I mentioned in my previous comment, we do not have to bother with builders in the `docs/Makefile`. If we decide this `Makefile` takes care of `generate` targets and nothing more, we can 1. move the `html` & co. from the `Makefile` into `pyproject.toml`, as you did, and then 2. reduce the `Makefile significantly, removing these "building foo" targets from it.

Reply:
Sorry I didn't get the update on the phone about the comments until after I replied. Indeed the approach you mentioned sounds like a good approach to get the best of both worlds right now. I'll handle that tomorrow.

It would be nice to eventually have custom domains (btw a helpful reference for that I think it's MATLAB's), but yeah that will take some more effort.